unstructured/partition/new_msg.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import annotations
```

Comment:
This is mentioned in the PR summary text above:
> **Additional Context**
> For reviewability purposes, this PR temporarily places the new partition_msg() implementation in new_msg.py and references that implementation from msg.py. new_msg.py will be renamed to msg.py in a closely following PR. This avoids a very messy interleaving of hunks in a diff between the old and re-written partition_msg() implementation.

Since there are only a few lines of the original left, I found it a lot easier to review like : "Here's what it was, that's all gone now", and "Here's the new version". I figured other reviewers would too.

The plan is to follow this PR immediately with one that changes `new_msg.py` into `msg.py`. The reviewability concern was the only driver here.

Reply:
Gotcha, thanks! Shoulda read the description closer haha